tests/components/zha/test_config_flow.py

```diff
@@ -1,125 +1,237 @@
-        homeassistant.components.zha.core.registries.RADIO_TYPES, new_radios, clear=True
-    ):
-        assert not await config_flow.check_zigpy_connection(
-            mock.sentinel.usb_path, mock.sentinel.unk_radio, mock.sentinel.zigbee_db
+async def test_user_port_config_fail(hass):
+    """Test port config flow."""
+    app_ctrl_cls = MagicMock()
+    app_ctrl_cls.SCHEMA_DEVICE = zigpy.config.SCHEMA_DEVICE
+    app_ctrl_cls.probe = AsyncMock(return_value=False)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: "pick_radio"}, data={CONF_RADIO_TYPE: "ezsp"}
+    )
+
+    with patch.dict(config_flow.RADIO_TYPES, {"ezsp": {CONTROLLER: app_ctrl_cls}}):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={zigpy.config.CONF_DEVICE_PATH: "/dev/ttyUSB33"},
         )
-        assert mock_radio.connect.call_count == 0
-        assert bad_radio.connect.call_count == 0
-        assert mock_ctrl.startup.call_count == 0
-        assert mock_ctrl.shutdown.call_count == 0
-
-        # unsuccessful radio connect
-        assert not await config_flow.check_zigpy_connection(
-            mock.sentinel.usb_path, mock.sentinel.bad_radio, mock.sentinel.zigbee_db
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "port_config"
+        assert result["errors"]["base"] == "cannot_connect"
+
+
+@pytest.mark.parametrize(
+    "radio_type, orig_ctrl_cls",
+    ((name, r[CONTROLLER]) for name, r in RADIO_TYPES.items()),
+)
+@patch("homeassistant.components.zha.async_setup_entry", AsyncMock(return_value=True))
+async def test_user_port_config(hass, radio_type, orig_ctrl_cls):
+    """Test port config."""
+    app_ctrl_cls = MagicMock()
+    app_ctrl_cls.SCHEMA_DEVICE = orig_ctrl_cls.SCHEMA_DEVICE
+    app_ctrl_cls.probe = AsyncMock(return_value=True)
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={CONF_SOURCE: "pick_radio"}, data={CONF_RADIO_TYPE: radio_type}
+    )
+
+    with patch.dict(
+        config_flow.RADIO_TYPES,
+        {radio_type: {CONTROLLER: app_ctrl_cls, "radio_description": "radio"}},
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={zigpy.config.CONF_DEVICE_PATH: "/dev/ttyUSB33"},
         )
-        assert mock_radio.connect.call_count == 0
-        assert bad_radio.connect.call_count == 1
-        assert mock_ctrl.startup.call_count == 0
-        assert mock_ctrl.shutdown.call_count == 0
-
-        # successful radio connect
-        assert await config_flow.check_zigpy_connection(
-            mock.sentinel.usb_path, mock.sentinel.radio, mock.sentinel.zigbee_db
+
+        assert result["type"] == "create_entry"
+        assert result["title"].startswith("/dev/ttyUSB33")
+        assert (
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)